File: goodReads.client/src/models/Follow.js
import { getDate } from "../utils/GetDate.js";
import { Account } from "./Account";

export class Follow {
  constructor(data) {
    this.id = data.id;
    this.creatorId = data.creatorId;
    this.followingUserId = data.followingUserId;
    this.createdAt = getDate(data.createdAt)
    this.updatedAt = getDate(data.updatedAt)
    // populated
    this.profile = new Account(data.profile);
  }

}

File: goodReads/Models/Follow.cs
namespace goodReads.Models;

public class Follow : IDbItem<int>
{

  public int Id { get; set; }
  public string creatorId { get; set; }

  public string FollowingUserId { get; set; }
  // public Profile FollowingUser { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }

  public Profile Profile { get; set; }
}


// public class Follow : Follow
// {
//   // public Profile Profile { get; set; }
// }
